### PR TITLE
chore: audit data visualisation challenges

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #XXXXX

Audits the D3 challenges for formatting changes to streamline the i18n process.

### CODE REVIEW

File: curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-scatterplot-graph.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ You can use HTML, JavaScript, CSS, and the D3 svg-based visualization library. T
 
 **User Story #4:** I can see dots, that each have a class of `dot`, which represent the data being plotted.
 
-**User Story #5:** Each dot should have the properties `data-xvalue` and `data-yvalue` containing their corresponding x and y values.
+**User Story #5:** Each dot should have the properties `data-xvalue` and `data-yvalue` containing their corresponding `x` and `y` values.
 
 **User Story #6:** The `data-xvalue` and `data-yvalue` of each dot should be within the range of the actual data and in the correct data format. For `data-xvalue`, integers (full years) or Date objects are acceptable for test evaluation. For `data-yvalue` (minutes), use Date objects.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-scatterplot-graph.md:27]:
```suggestion
**User Story #6:** The `data-xvalue` and `data-yvalue` of each dot should be within the range of the actual data and in the correct data format. For `data-xvalue`, integers (full years) or `Date` objects are acceptable for test evaluation. For `data-yvalue` (minutes), use `Date` objects.
```

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/add-document-elements-with-d3.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ The `body` should have one `h1` element.
 assert($('body').children('h1').length == 1);
 ```
 
-The `h1` element should have the text "Learning D3" in it.
+The `h1` element should have the text `Learning D3` in it.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/add-document-elements-with-d3.md:47]: Line 37 of this file needs `"Learning D3"` to be in backticks. _I wish GitHub let  you make further suggestions_

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-styles-based-on-data.md [hunk 3]
```diff
@@ -22,10 +22,9 @@ The `style()` method is not limited to setting the `color` - it can be used with
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-styles-based-on-data.md:22]: Line 17 comment in example code. Is this in the dictionary?

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-the-color-of-an-svg-element.md [hunk 4]
```diff
@@ -14,7 +14,7 @@ In SVG, a `rect` shape is colored with the `fill` attribute. It supports hex cod
 
 # --instructions--
 
-Add an `attr()` method to set the "fill" of all the bars to the color "navy".
+Add an `attr()` method to set the `fill` of all the bars to the color `navy`.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-the-color-of-an-svg-element.md:17]: Here, `navy` is within backticks, but the names of the colors are not, in the `hints` section.

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-styles-based-on-data.md [hunk 5]
```diff
@@ -22,10 +22,9 @@ The `style()` method is not limited to setting the `color` - it can be used with
...
 # --instructions--
 
-Add the `style()` method to the code in the editor to set the `color` of the `h2` elements conditionally. Write the callback function so if the data value is less than 20, it returns "red", otherwise it returns "green".
+Add the `style()` method to the code in the editor to set the `color` of the `h2` elements conditionally. Write the callback function so if the data value is less than 20, it returns `red`, otherwise it returns `green`.
 
-**Note**  
-You can use if-else logic, or the ternary operator.
+**Note:** You can use if-else logic, or the ternary operator.
 
 # --hints--
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-styles-based-on-data.md:29]: The names of the colours here (below) are not in backticks. Should they be?

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Only did the first 15.

File: curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-scatterplot-graph.md (hunk 1 above)
Comment (ShaunSHamilton) [reply curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-scatterplot-graph.md:27]: My thinking here was: This is a non-translatable reference to the JS `Date` object. That is, it requires someone to write the code `Date`.

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/add-document-elements-with-d3.md (hunk 2 above)
Comment (nhcarrigan) [reply curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/add-document-elements-with-d3.md:47]: > I wish GitHub let you make further suggestions

ME TOOOOOO

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-styles-based-on-data.md (hunk 3 above)
Comment (nhcarrigan) [reply curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-styles-based-on-data.md:22]: Nope - I just missed it 😅

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-the-color-of-an-svg-element.md (hunk 4 above)
Comment (nhcarrigan) [reply curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-the-color-of-an-svg-element.md:17]: So this is actually a good question.

The instructions read like we expect the camper to use "color: navy", but the test checks the actual rgb value so it's not 100% necessary to use `navy`. I could go either way on this one.

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-styles-based-on-data.md (hunk 5 above)
Comment (nhcarrigan) [reply curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-styles-based-on-data.md:29]: Same thing here - I'm more than happy to wrap them in backticks, but should that be done when the test doesn't look for the colour string specifically? 🤔

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-the-color-of-an-svg-element.md (hunk 4 above)
Comment (RandellDawson) [reply curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-the-color-of-an-svg-element.md:17]: If the user specifies the color "navy" like `.attr("fill", "navy");` or `rgb(0, 0, 128)`, will the test pass?  If so, then I would remove the backticks and just use the word **navy** (not necessarily bold).

Comment (nhcarrigan) [reply curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-the-color-of-an-svg-element.md:17]: Correct - it passes if I use `.attr("fill", "rgb(0,0,128)")`.

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-styles-based-on-data.md (hunk 5 above)
Comment (RandellDawson) [reply curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/change-styles-based-on-data.md:29]: Based on the reply to the other question I posed above, I suggest leaving the color names without backticks for these two instances.

File: curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/get-geolocation-data-to-find-a-users-gps-coordinates.md [hunk 6]
```diff
@@ -52,7 +52,7 @@ Your code should use `position.coords.longitude` to display the user's longitudi
 assert(code.match(/position\.coords\.longitude/g));
 ```
 
-You should display the user's position within the `data` div element.
+You should display the user's position within the `data` `div` element.
```
Comment (RandellDawson) [inline curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/get-geolocation-data-to-find-a-users-gps-coordinates.md:55]:
```suggestion
You should display the user's position within the `div` element with `id="data"`.
```

**Review by RandellDawson — DISMISSED** [review]
LGTM

@ShaunSHamilton If you are fine with the current state of this PR, feel free to merge.

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Just a couple things

**Comment by ShaunSHamilton** [comment]
Nicholas/ @RandellDawson I realise our main aim is to just ensure what the tests rely on is not translated. So, if this is off-topic, then it can be something for another discussion:

There are many references to very specific named coding elements (e.g. `JSON`, `API`), which are not wrapped in backticks. Should they be? I see we often wrap `POST`, but not `JSON`.

**Comment by nhcarrigan** [comment]
`POST` is a word that has translations, where `API` and `JSON` are acronyms and wouldn't be translated accidentally, I believe.

**Review by ShaunSHamilton — APPROVED** [review]
I am happy with this as is 👍 

Thanks, Nicholas.

**Review by RandellDawson — APPROVED** [review]

**Pull request merged**